src/librustc_resolve/late.rs

```diff
@@ -1041,9 +1018,8 @@ impl<'a, 'b> LateResolutionVisitor<'a, '_> {
                               impl_items: &[ImplItem]) {
         debug!("resolve_implementation");
         // If applicable, create a rib for the type parameters.
-        self.with_generic_param_rib(HasGenericParams(generics, ItemRibKind {
-            suggest_adding_parameter: true,
-        }), |this| {
+        self.with_generic_param_rib(generics, ItemRibKind { suggest_adding_parameter: true },
```

Comment:
Consider moving `ItemRibKind { ... }` to a `let rib_kind` binding to make the `|this| {` look slightly less weird.

Reply:
Fixed.